cmd/query/app/querysvc/query_service_test.go

```diff
@@ -59,42 +59,63 @@ var (
 	}
 )
 
-func initializeTestServiceWithArchiveOptions() (*QueryService, *spanstoremocks.Reader, *depsmocks.Reader, *spanstoremocks.Reader, *spanstoremocks.Writer) {
-	readStorage := &spanstoremocks.Reader{}
-	dependencyStorage := &depsmocks.Reader{}
-	archiveReadStorage := &spanstoremocks.Reader{}
-	archiveWriteStorage := &spanstoremocks.Writer{}
-	options := QueryServiceOptions{
-		ArchiveSpanReader: archiveReadStorage,
-		ArchiveSpanWriter: archiveWriteStorage,
+type testMocks struct {
+	spanReader *spanstoremocks.Reader
+	depsReader *depsmocks.Reader
+
+	archiveSpanReader *spanstoremocks.Reader
+	archiveSpanWriter *spanstoremocks.Writer
+}
+
+type QueryServiceOptionApplier func(*testMocks, *QueryServiceOptions)
+
+func withArchiveSpanReader() QueryServiceOptionApplier {
+	return func(mocks *testMocks, options *QueryServiceOptions) {
+		r := &spanstoremocks.Reader{}
+		options.ArchiveSpanReader = r
+		mocks.archiveSpanReader = r
 	}
-	qs := NewQueryService(readStorage, dependencyStorage, options)
-	return qs, readStorage, dependencyStorage, archiveReadStorage, archiveWriteStorage
 }
 
-func initializeTestServiceWithAdjustOption() *QueryService {
-	readStorage := &spanstoremocks.Reader{}
-	dependencyStorage := &depsmocks.Reader{}
-	options := QueryServiceOptions{
-		Adjuster: adjuster.Func(func(trace *model.Trace) (*model.Trace, error) {
+func withArchiveSpanWriter() QueryServiceOptionApplier {
+	return func(mocks *testMocks, options *QueryServiceOptions) {
+		r := &spanstoremocks.Writer{}
+		options.ArchiveSpanWriter = r
+		mocks.archiveSpanWriter = r
+	}
+}
+
+func withAdjuster() QueryServiceOptionApplier {
+	return func(mocks *testMocks, options *QueryServiceOptions) {
+		options.Adjuster = adjuster.Func(func(trace *model.Trace) (*model.Trace, error) {
 			return trace, errAdjustment
-		}),
+		})
 	}
-	qs := NewQueryService(readStorage, dependencyStorage, options)
-	return qs
 }
 
-func initializeTestService() (*QueryService, *spanstoremocks.Reader, *depsmocks.Reader) {
+func initializeTestService(optionAppliers ...QueryServiceOptionApplier) (*QueryService, *testMocks) {
```

Comment:
since you're already introducing a new struct testMocks, you might instead return a single `testQueryService` struct which contains both the `*QS` and the mocks at top level

Reply:
glad you mentioned this as this was also what I wanted to do initially, just thought there may have been a preference to separate the "thing under test" from the mocks.

Reply:
tqsOption - these are private types, no need to be especially creative